app/uk/gov/hmrc/customs/declaration/services/CustomsAuthService.scala

```diff
@@ -0,0 +1,164 @@
+        logger.error("Error authorising CSP", e)
+        Left(ErrorInternalServerError)
+    }
+  }
+
+  /*
+  Wrapper around HMRC authentication library authorised function for NON CSP authentication
+  */
+  def authAsNonCsp[A](withNrs: Boolean)(implicit vhr: ValidatedHeadersRequest[A], hc: HeaderCarrier): Future[Either[ErrorResponse, NonCsp]] = {
+    val eventualAuth: Future[(Enrolments, Option[NrsRetrievalData])] =
+      if (withNrs) {
+        authorised(Enrolment("HMRC-CUS-ORG") and AuthProviders(GovernmentGateway)).retrieve(nonCspRetrievals) {
+          case internalId ~ externalId ~ agentCode ~ credentials ~ confidenceLevel ~ nino ~ saUtr ~ name ~ dateOfBirth ~ email ~ agentInformation ~ groupIdentifier ~
+            credentialRole ~ mdtpInformation ~ itmpName ~ itmpDateOfBirth ~ itmpAddress ~ affinityGroup ~ credentialStrength ~ loginTimes ~ authorisedEnrolments =>
+
+            val retrievalData = NrsRetrievalData(internalId, externalId, agentCode, credentials, confidenceLevel, nino, saUtr,
+              name, dateOfBirth, email, agentInformation, groupIdentifier, credentialRole, mdtpInformation, itmpName,
+              itmpDateOfBirth, itmpAddress, affinityGroup, credentialStrength, loginTimes)
+
+            Future.successful((authorisedEnrolments, Some(retrievalData)))
+        }
+      } else {
+        authorised(Enrolment("HMRC-CUS-ORG") and AuthProviders(GovernmentGateway)).retrieve(Retrievals.authorisedEnrolments) {
+          enrolments =>
+            Future.successful((enrolments, None))
+        }
+      }
+
+    eventualAuth.map{ enrolmentsAndMaybeNrsData =>
+      val enrolments: Enrolments = enrolmentsAndMaybeNrsData._1
+      val maybeNrsData: Option[NrsRetrievalData] = enrolmentsAndMaybeNrsData._2
+      val maybeEori: Option[Eori] = findEoriInCustomsEnrolment(enrolments, hc.authorization)
+      logger.debug(s"EORI from Customs enrolment for non-CSP request: $maybeEori")
+      maybeEori.fold[Either[ErrorResponse, NonCsp]]{
+        googleAnalyticsConnector.failure(errorResponseEoriNotFoundInCustomsEnrolment.message)
+        Left(errorResponseEoriNotFoundInCustomsEnrolment)
+      }{ eori =>
+        logger.debug("Authorising as non-CSP")
+        Right(NonCsp(eori, maybeNrsData))
+      }
+    }.recover{
+      case NonFatal(_: AuthorisationException) =>
+        googleAnalyticsConnector.failure(errorResponseUnauthorisedGeneral.message)
+        Left(errorResponseUnauthorisedGeneral)
+      case NonFatal(e) =>
+        logger.error("Error authorising non-CSP", e)
+        Left(ErrorInternalServerError)
+    }
+  }
+
+  private def findEoriInCustomsEnrolment[A](enrolments: Enrolments, authHeader: Option[Authorization])(implicit vhr: ValidatedHeadersRequest[A], hc: HeaderCarrier): Option[Eori] = {
+    val maybeCustomsEnrolment = enrolments.getEnrolment("HMRC-CUS-ORG")
+    if (maybeCustomsEnrolment.isEmpty) {
+      logger.warn(s"Customs enrolment HMRC-CUS-ORG not retrieved for authorised non-CSP call")
```

Comment:
HMRC-CUS-ORG can be extracted into a val now that its being used in a few places

Reply:
done